### PR TITLE
Add `misalignment` property to `BPM`

### PR DESCRIPTION
**Pull request by roussel-ryan**
## Description

Adds a misalignment parameter for the `BPM` class.

## Motivation and Context

Enables misalignments with respect to the reference axis to be specified for BPMs. Needed, for example, to identify the misalignment of BPMs.

- [ ] I have raised an issue to propose this change (required for new features and bug fixes)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

- [ ] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (_required for a bug fix or a new feature_).
- [x] I have updated the documentation accordingly.
  
  
- [ ] I have reformatted the code and checked that formatting passes (**required**).
- [ ] I have have fixed all issues found by `flake8` (**required**).
- [ ] I have ensured that all `pytest` tests pass (**required**).
- [ ] I have run `pytest` on a machine with a CUDA GPU and made sure all tests pass (**required**).
- [ ] I have checked that the documentation builds (**required**).

Note: We are using a maximum length of 88 characters per line.

### CODE REVIEW

**Review by jank324 — CHANGES_REQUESTED** [review]
The changelog entry is missing

File: cheetah/accelerator/bpm.py [hunk 1]
```diff
@@ -34,13 +36,22 @@ def __init__(
...
         self.register_buffer(
             "reading",
             torch.as_tensor((torch.nan, torch.nan), device=device, dtype=dtype),
             persistent=False,
         )
 
+        self.register_buffer_or_parameter(
+            "misalignment",
+            torch.as_tensor(
+                misalignment if misalignment is not None else (0.0, 0.0),
```
Comment (jank324) [inline cheetah/accelerator/bpm.py:50]:
```suggestion
                misalignment if misalignment is not None else torch.zeros(2, device=device, dtype=dtype),
```
